wordpress-mobile/WordPress-Android
# Issue/14128 reader related posts UI refactoring

Task #14128

This PR is a sub-task of Issue #14128  and refactors/ restructures related posts UI logic. It 

1. Builds `RelatedPostsUiState` in `ReaderPostDetailsUiStateBuilder` for the related posts. 
2. Displays related posts using a recycler view.

To test:
1. Open a wp.com reader post. 
2. Notice that local related posts (from selected post's site) are shown at the bottom of the post as recycler view cards with a featured image and title.
3. Notice that global related posts (from another wp.com site) are shown at the bottom of the post as recycler view cards with  featured image, title and follow button.

Tests included in `ReaderPostDetailsUiStateBuilderTest`.

Notes:

1. Related posts actions are implemented in #14163.
2. Ignore all stying. UI fine-tuning + cleanup (related to `ReaderSimplePost` + `ReaderSimplePostView`) will be done separately. 
3. Yet to set featured image size.

UI Design:

Waiting for 
- Follow button positioning for global posts.
- Whether to show just the title and no image if featured image is not set for a post. I haven't removed redundant fields from  `ReaderSimplePost` until design is finalised (in case we decide to show some other fields).

PR submission checklist:
